pkgs/development/libraries/db/generic.nix

```diff
@@ -19,10 +22,48 @@ stdenv.mkDerivation (rec {
     sha256 = sha256;
   };
 
+  # The provided configure script features `main` returning implicit `int`, which causes
+  # configure checks to work incorrectly with clang 16.
+  nativeBuildInputs = lib.optionals stdenv.cc.isClang [ autoreconfHook ];
```

Comment:
Can't we just reconfigure on all platforms? It would make things more uniform and bugs would be easier to spot on linux.

Reply:
I can see if it works on Linux and open a PR to make it unconditional if there are no issues.

Reply:
That would be awesome!

Reply:
I opened #242060 to make them unconditional.